library.rb

```diff
@@ -48,12 +40,7 @@ def most_popular_book(quantity)
   end
 
   def how_many_people_ordered_one_of_three_most_popular_books
-    best_books = most_popular_book(3)
-    count_people = {}
-    best_books.each do |book|
-      count_people[book.to_s] = @orders.find_all { |order| order.book.to_s == book.to_s }.count
-    end
-    count_people.each {|book, count| "#{book}:#{count}"}
+    most_popular_book(3).map { |book| @orders.find_all { |order| order.book.to_s == book.to_s }.group_by(&:reader).uniq }.uniq.count
```

Comment:
most_popular_book(3) already returns books with orders no need to select from all orders again. You can use map and flatten to select orders from most_popular_book(3) result and then group by reader and count (grouped values already uniq)